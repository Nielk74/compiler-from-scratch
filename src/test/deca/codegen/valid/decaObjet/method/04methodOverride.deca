// Description:
//      Override method
//
// Resultats:
//    04methodOverride.deca
//
// Historique:
//    cree le 16/06/2022

class Parent {
    void m1()
    {
        println("From parent m1()");
    }
  
    void m2()
    {
        println("From parent m2()");
    }
}
  
class Child extends Parent {
    // new m1() method
    // unique to Child class
    void m1()
    {
        println("From child m1()");
    }
  

}

class GrandChild extends Child{
    void m2()
    {
        println("From grand child m2()");
    }
}


{
    Parent parent = new Parent();
    Child child = new Child();
    GrandChild grandChild = new GrandChild();
    parent.m1();
    parent.m2();
    child.m1();
    child.m2();
    grandChild.m1();
    grandChild.m2();
}