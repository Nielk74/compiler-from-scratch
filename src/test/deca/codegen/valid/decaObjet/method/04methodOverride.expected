From parent m1()
From parent m2()
From child m1()
From parent m2()
From child m1()
From grand child m2()
